src/nvim/map.c

```diff
@@ -123,29 +123,8 @@
     kh_clear(T##_##U##_map, map->table); \
   }
 
-static inline khint_t String_hash(String s)
```

Comment:
I don't mind removing the indirection, even if we have helper functions/macros they can just operate on the khash types directly. But anyway I guess this could be in `khash.c` as it is khash that receives a hash function.